src/components/Pin.vue

```diff
@@ -0,0 +1,38 @@
+<template>
+  <div class="pin-container"
+       @mouseover="isHovered = true"
+       @mouseout="isHovered = false"
+       @click="changeImage">
+    <img :src="pinUrl" alt="Pin" :style="{ transform: isHovered ? 'scale(0.9)' : 'scale(1)', transition: 'transform 0.3s ease', width: '75px', height: '100px' }">
+  </div>
+</template>
+
+<script setup lang="ts">
+import { ref } from 'vue';
+
+const pinUrls = ['src/assets/pin/Pin_Azul.png', 
+                'src/assets/pin/Pin_Verde.png', 
+                'src/assets/pin/Pin_Amarillo.png', 
+                'src/assets/pin/Pin_Marron.png', 
+                'src/assets/pin/Pin_Morado.png', 
+                'src/assets/pin/Pin_Rojo.png'];
```

Comment:
Esto así tal cual no va a funcionar en producción. El motivo es que el bundler añade al nombre de cada archivo su hash para hacer [cache busting](https://javascript.plainenglish.io/what-is-cache-busting-55366b3ac022)

Por lo tanto, el nombre que tienen en local != al nombre que van a tener durante producción. [Esto se arregla así](https://vitejs.dev/guide/assets.html#importing-asset-as-url). Además, esto mantiene una referencia a los archivos en tiempo de construcción y si dejan de existir o estan corruptos nos enteraremos!

Sería algo asi (cambio ``pinUrls`` por ``pinImgs`` porque técnicamente son imágenes):

```
import Amarillo from '@/assets/pin/Pin_Amarillo.png';
import ...
...

const pinImgs = [Amarillo, ...];
```